.ci/containers/downstream-builder/generate_downstream.sh

```diff
@@ -42,6 +56,12 @@ function clone_repo() {
         exit 1
     fi
 
+    # the check of existence of BASE_BRANCH is for backwards-compatability
+    if [ -n "$BASE_BRANCH" ]; then
+        echo "BASE_BRANCH: $BASE_BRANCH"
+        git -C $LOCAL_PATH checkout $BASE_BRANCH
```

Comment:
Would this work? It seems like the `git clone` might have to happen before checking out a branch.

FWIW you might also be able to use `git clone $GITHUB_PATH $LOCAL_PATH --branch $BASE_BRANCH` if there's a default value for BASE_BRANCH.